mindmeld/components/dialogue.py

```diff
@@ -972,14 +988,22 @@ def __call__(self, request, responder):
             request (Request): The request object.
             responder (DialogueResponder): The responder object.
         """
+
+        # If form iteration in request object, continue using that.
+        # If None, set to original form.
+        if request.form:
+            self._local_entity_form = request.form["entities"] or None
+        else:
+            self._local_entity_form = None
+
         if request.text.lower() in self._exit_keys:
             responder.reply(self._exit_response)
             self._exit_flow(responder)
             return
 
-        self._set_next_turn(request, responder)
+        responder = self._set_next_turn(request, responder)
```

Comment:
Why do we assign back to responder here while in the past, we didnt?

Reply:
We don't need to do this, because the responder object is carried in the class, but just for code completeness I added this. Can revert this.

Reply:
Reverted.